With the changes in this PR, mypy complains:
```
homeassistant/data_entry_flow.py:366: error: "object" not callable  [operator]
Found 1 error in 1 file (checked 10 source files)
```

--- a/homeassistant/components/config/config_entries.py
+++ b/homeassistant/components/config/config_entries.py
@@ -141,7 +141,9 @@ def _prepare_config_flow_result_json(
     return data
 
 
-class ConfigManagerFlowIndexView(FlowManagerIndexView):
+class ConfigManagerFlowIndexView(
+    FlowManagerIndexView[config_entries.ConfigEntriesFlowManager]
+):
     """View to create config flows."""
 
     url = "/api/config/config_entries/flow"
@@ -196,7 +198,9 @@ class ConfigManagerFlowIndexView(FlowManagerIndexView):
         return _prepare_config_flow_result_json(result, super()._prepare_result_json)
 
 
-class ConfigManagerFlowResourceView(FlowManagerResourceView):
+class ConfigManagerFlowResourceView(
+    FlowManagerResourceView[config_entries.ConfigEntriesFlowManager]
+):
     """View to interact with the flow manager."""
 
     url = "/api/config/config_entries/flow/{flow_id}"
@@ -238,7 +242,9 @@ class ConfigManagerAvailableFlowView(HomeAssistantView):
         return self.json(await async_get_config_flows(hass, **kwargs))
 
 
-class OptionManagerFlowIndexView(FlowManagerIndexView):
+class OptionManagerFlowIndexView(
+    FlowManagerIndexView[config_entries.OptionsFlowManager]
+):
     """View to create option flows."""
 
     url = "/api/config/config_entries/options/flow"
@@ -255,7 +261,9 @@ class OptionManagerFlowIndexView(FlowManagerIndexView):
         return await super().post(request)
 
 
-class OptionManagerFlowResourceView(FlowManagerResourceView):
+class OptionManagerFlowResourceView(
+    FlowManagerResourceView[config_entries.OptionsFlowManager]
+):
     """View to interact with the option flow manager."""
 
     url = "/api/config/config_entries/options/flow/{flow_id}"

--- a/homeassistant/data_entry_flow.py
+++ b/homeassistant/data_entry_flow.py
@@ -363,7 +363,7 @@ class FlowManager(abc.ABC, Generic[_FlowResultT]):
             data_schema := cur_step.get("data_schema")
         ) is not None and user_input is not None:
             try:
-                user_input = data_schema(user_input)
+                user_input = data_schema(user_input)  # type: ignore[operator]
             except vol.Invalid as ex:
                 raised_errors = [ex]
                 if isinstance(ex, vol.MultipleInvalid):

--- a/homeassistant/helpers/data_entry_flow.py
+++ b/homeassistant/helpers/data_entry_flow.py
@@ -2,9 +2,10 @@
 from __future__ import annotations
 
 from http import HTTPStatus
-from typing import Any
+from typing import Any, Generic
 
 from aiohttp import web
+from typing_extensions import TypeVar
 import voluptuous as vol
 import voluptuous_serialize
 
@@ -14,11 +15,17 @@ from homeassistant.components.http.data_validator import RequestDataValidator
 
 from . import config_validation as cv
 
+_FlowManagerT = TypeVar(
+    "_FlowManagerT",
+    bound=data_entry_flow.FlowManager[Any],
+    default=data_entry_flow.FlowManager,
+)
 
-class _BaseFlowManagerView(HomeAssistantView):
+
+class _BaseFlowManagerView(HomeAssistantView, Generic[_FlowManagerT]):
     """Foundation for flow manager views."""
 
-    def __init__(self, flow_mgr: data_entry_flow.FlowManager) -> None:
+    def __init__(self, flow_mgr: _FlowManagerT) -> None:
         """Initialize the flow manager index view."""
         self._flow_mgr = flow_mgr
 
@@ -48,7 +55,7 @@ class _BaseFlowManagerView(HomeAssistantView):
         return data
 
 
-class FlowManagerIndexView(_BaseFlowManagerView):
+class FlowManagerIndexView(_BaseFlowManagerView[_FlowManagerT]):
     """View to create config flows."""
 
     @RequestDataValidator(
@@ -96,7 +103,7 @@ class FlowManagerIndexView(_BaseFlowManagerView):
         return {"show_advanced_options": data["show_advanced_options"]}
 
 
-class FlowManagerResourceView(_BaseFlowManagerView):
+class FlowManagerResourceView(_BaseFlowManagerView[_FlowManagerT]):
     """View to interact with the flow manager."""
 
     async def get(self, request: web.Request, /, flow_id: str) -> web.Response:

